code/feature_extraction/ImageHelper.py

```diff
@@ -3,26 +3,34 @@
 except ImportError:
     import Image
 import pytesseract
-import os
+from pathlib import Path
 
 # Make sure that Tesseract is defined even if it is not in $PATH variable
 pytesseract.pytesseract.tesseract_cmd = r'C:\Program Files\Tesseract-OCR\tesseract'
 
 
 class ImageHelper:
-    @staticmethod
-    def get_text(image_path):
-        # Check if path is not a list from pandas
-        if not isinstance(image_path, str):
-            image_path = image_path[0]
+    def __init__(self, data_path):
+        project_path = Path(__file__).parent.parent
+        data_path = (project_path / data_path)
 
-        # RPath is parent of currentpath (so the appliedNLP folder)
-        rpath = os.path.realpath("..")
+        self.data_path = data_path
 
-        # Join paths
-        # TODO: this is very ugly and surely can be done easier
-        path = os.path.join(rpath, "data", "clickbait17-train-170331", image_path.split("/")[0],
-                            image_path.split("/")[1])
+    def get_text(self, image_path):
+        # Check if input is not a list from pandas
+        if not isinstance(image_path, str):
+            if len(image_path) == 0:
+                image_path = ""
+            elif len(image_path) == 1:
+                image_path = image_path[0]
+            else:
+                # Article has multiple pictures: this should not occur
+                image_path = ""
 
-        return pytesseract.image_to_string(Image.open(path))
+        # Return empty string if image path was undefined
+        if image_path == "":

```

Comment:
also check for `is None` (and put as the first line in this function to fail fast)